x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/ServicePage/CancelButton.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButtonEmpty, EuiButtonEmptyColor } from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { history } from '../../../../../../utils/history';
+
+export function CancelButton({
+  color = 'primary'
+}: {
+  color?: EuiButtonEmptyColor;
+}) {
+  return (
+    <EuiButtonEmpty
+      color={color}
+      onClick={() => {
+        history.push({
+          pathname: '/settings/agent-configuration',
+          search: history.location.search
+        });
```

Comment:
This seemed like the easiest way to link to the route. But I'd really prefer something like:
```js
navigateToRoute('/settings/agent-configuration');
```

where `navigateToRoute` is based on the existing routes and will take their (query and path) params into account.

Reply:
I'm using `APMLink`.
```js
<APMLink path={`/settings/customize-ui`}>
       <EuiButtonIcon
          iconType="gear"
          color="text"
          aria-label="Custom links settings page"
       />
</APMLink>
```

Reply:
If you have unsaved changes (which is the current behaviour), wouldn't it make more sense to cancel the edits and return to the create page where you started instead of sending the user back to the list page?